solutions/USACO dec19-whereami.mdx

```diff
@@ -0,0 +1,46 @@
+---
+id: usaco-964
+source: USACO Bronze 2019 December
+title: Where Am I?
+author: Ananth Kashyap
+---
+
+[Official Analysis](http://www.usaco.org/current/data/sol_whereami_bronze_dec19.html)
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N^2)$
+
+<LanguageSection>
+
+<PySection>
+
+```python
+# Take in input using Python file i/o system
+file_in = open('whereami.in')
+data = file_in.read().strip().split('\n')
+n = int(data[0])
+mailboxes = data[1]
+
+# Set the answer initially to n, as we know n is always a possible answer
+ans = n
+
+# We can iterate through lengths of sequences to find the smallest length
+for l in range(1, n + 1):
+	# Store all sequences of length n in a set
+	# Sets only hold distinct elements
+	sequences = set()
+	for i in range(n - l + 1):
+		sequences.add(mailboxes[i : i + l])
+	# Check if all sequences are unique
```

Comment:
```suggestion
	# Check if all substrings are unique by checking whether the length of the set is equal to the total number of sequences for a fixed l
```

Reply:
keep in mind line length; you might have to break it into two lines

Reply:
```suggestion
	# Check if all substrings are unique
```

This is sufficient. These are just comments in code, not a solution explanation.